### PR TITLE
Feat/Update task

### PR DESCRIPTION
**Pull request by TIEN1909**
#177  Implement handle update task

**Result**

> Responsive screens

- X-Small devices (portrait phones, from 375px)

- Initial state
image

- After handle
image

- Small devices (landscape phones, 576px and up)

- Initial state
image

- After handle
image

### CODE REVIEW

File: javascript-training/big-practice/pomodoro-app/src/index.html [hunk 1]
```diff
@@ -39,7 +39,7 @@ <h1 class="title capitalize">tasks</h1>
 
     <section class="add-task container">
 
-      <form class="form-add-task hidden">
+      <form class="form-add-task ">
```
Comment (hungdo-agilityio) [inline javascript-training/big-practice/pomodoro-app/src/index.html:42]: Remove extra space at the end, please check for all

File: javascript-training/big-practice/pomodoro-app/src/scripts/controllers/index.js [hunk 2]
```diff
@@ -1,26 +1,65 @@
...
+  }
+
+  /**
+   * Handle the event when adding or updating a task.
+   * @param {string|null} taskId - The ID of the task (null if adding a new task).
+   * @param {string} taskName - The task name.
+   * @param {number} numberPomodoro - The number of Pomodoros.
+   * @param {boolean} isUpdate - True if updating a task, false if adding a new task.
+   */
+  handleSubmitTask(taskId, taskName, numberPomodoro, isUpdate) {
```
Comment (hungdo-agilityio) [inline javascript-training/big-practice/pomodoro-app/src/scripts/controllers/index.js:32]: Please help to check order parameters

File: javascript-training/big-practice/pomodoro-app/src/scripts/templates/index.js [hunk 3]
```diff
@@ -2,21 +2,21 @@ import editIcon from '../../assets/icons/icon-edit.png';
...
         <div class="task-item-done">
           <span class="task-actual">0</span>
           <span>/</span>
-          <span class="task-est">${task.inputValue}</span>
+          <span class="task-est">${newTask.estPomodoro}</span>
         </div>
-        <button class="btn-more border-platinum pointer opacity">
-          <img src="${editIcon}" alt="icon more" class="icon-more edit">
+        <button class="btn-more btn-show-task border-platinum pointer opacity">
+          <img src="${editIcon}" alt="icon more" class="icon-more edit ">
```
Comment (tungtran-agilityio) [inline javascript-training/big-practice/pomodoro-app/src/scripts/templates/index.js:19]: remove space

File: javascript-training/big-practice/pomodoro-app/src/scripts/views/formView.js [hunk 4]
```diff
@@ -2,63 +2,96 @@ import { helpers } from '../helpers/index';
...
 
-  // Toggle the display of the task form when clicking the "Add Task" button
   toggleTaskForm() {
-    this.btnAddTask.addEventListener('click', () => {
-      helpers.dom.toggleDisplay(this.taskForm, this.btnAddTask);
+    this.btnAddTaskElement.addEventListener('click', () => {
+      helpers.dom.toggleDisplay(this.taskFormElement, this.btnAddTaskElement);
+      const showForm = this.taskFormElement.classList.contains('hidden');
+
+      if (this.taskInputElement !== '' && showForm) {
```
Comment (hungdo-agilityio) [inline javascript-training/big-practice/pomodoro-app/src/scripts/views/formView.js:23]: Any chance for `this.taskInputElement` equal empty string

**Review by tungtran-agilityio — APPROVED** [review]

**Pull request merged**